File: mgmt-application/src/main/java/io/quartic/mgmt/MgmtConfiguration.java
package io.quartic.mgmt;

import io.dropwizard.Configuration;
import io.quartic.catalogue.api.DatasetNamespace;

public class MgmtConfiguration extends Configuration {
    private String catalogueUrl;
    private String bucketName;
    private String howlUrl;
    private DatasetNamespace defaultCatalogueNamespace;

    public String getCatalogueUrl() {
        return catalogueUrl;
    }

    public void setCatalogueUrl(String catalogueUrl) {
        this.catalogueUrl = catalogueUrl;
    }

    public String getBucketName() {
        return bucketName;
    }

    public void setBucketName(String bucketName) {
        this.bucketName = bucketName;
    }

    public String getHowlUrl() {
        return howlUrl;
    }

    public void setHowlUrl(String howlUrl) {
        this.howlUrl = howlUrl;
    }

    public DatasetNamespace getDefaultCatalogueNamespace() {
        return defaultCatalogueNamespace;
    }

    public void setHowlUrl(DatasetNamespace defaultCatalogueNamespace) {
        this.defaultCatalogueNamespace = defaultCatalogueNamespace;
    }
}

File: mgmt-application/src/main/kotlin/io/quartic/mgmt/MgmtConfiguration.kt
package io.quartic.mgmt

import io.quartic.catalogue.api.model.DatasetNamespace
import io.quartic.common.application.ConfigurationBase
import io.quartic.common.auth.User
import io.quartic.mgmt.auth.Multimap

data class GithubConfiguration(
    val oauthApiRoot: String = "https://github.com",
    val apiRoot: String = "https://api.github.com",
    val clientId: String,
    val clientSecret: String,
    val trampolineUrl: String,
    val scopes: List<String>,
    val redirectHost: String
)

data class CookiesConfiguration(
    val secure: Boolean,
    val maxAgeSeconds: Int
)

data class MgmtConfiguration(
    val catalogueUrl: String,
    val howlUrl: String,
    val registryUrl: String,
    val bildUrl: String,
    var github: GithubConfiguration,
    val cookies: CookiesConfiguration,
    val tokenTimeToLiveMinutes: Int = 60
): ConfigurationBase()
